### PR TITLE
feat(gui): pair the DPI slider with an editable numeric field

### LINKED ISSUE #1008: [Feature]: Readjust DPI Number Range / Rethink Slider
### Pre-flight checklist

- [x] I searched existing issues and the [Roadmap](https://github.com/AprilNEA/OpenLogi#roadmap), and this isn't already tracked.

### Problem / motivation

Most users don't use DPI settings next to 100 or 2600. 
This large range makes it awkward to use the slider for usable ranges. There needs to be a better option.

### Proposed solution

**Slider + numeric field combo**

Pair the slider with an editable number box. The slider handles quick/coarse adjustment in the common range; the text field handles precise or extreme values. This is probably the single most common real-world solution because it sidesteps the scaling problem entirely. power users type, casual users drag.

### Alternatives considered

_No response_

### Related area(s)

- [ ] GUI
- [ ] CLI
- [ ] Button actions / remapping
- [x] DPI
- [ ] SmartShift
- [ ] Per-application profiles
- [ ] Configuration (TOML)
- [ ] Auto-update
- [x] Other

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 4ni1ak**
## Summary

Pairs the DPI slider with a small digits-only text field, so an exact value like
1600 can be typed directly instead of dragged to across the device's full range in
one pixel-for-pixel sweep.

## Changes

**`crates/openlogi-desktop/src/features/pointer/dpi.rs`**

- Adds a numeric `InputState` field beside the slider, built/torn down alongside it
  (same lifecycle: appears only where the slider itself does — a fixed-DPI device
  still shows the plain value line).
- Commits through `AppState::normalize_active_dpi`, the exact path the slider's own
  Release handler already uses, so a typed value that isn't one of the device's
  supported steps snaps to the nearest one the same way dragging to it would.
- The field's displayed text tracks external changes (slider drags, device
  switches, capability reads) on every render, mirroring the slider thumb — except
  while the field is focused, so an update landing mid-edit can't overwrite what
  the user is typing.
- `parse_dpi_input` is a pure function with its own unit tests: empty/non-numeric
  input is a no-op rather than committing garbage, and a value too large for the
  wire format's `u16` saturates instead of panicking (reachable only by typing an
  implausibly long digit run, since the field's own validator already restricts
  input to digits).

No new user-facing strings were needed (the field carries no label), so the locale
catalogs are untouched.

## Testing

- `cargo fmt --all -- --check` — pass
- `cargo test -p openlogi-desktop i18n` — pass (parity test, unaffected but run for
  completeness since this touches GUI code)
- `cargo clippy -p openlogi-desktop --all-targets -- -D warnings` — pass
- `cargo test -p openlogi-desktop` — pass, 189 tests including 5 new ones covering
  `parse_dpi_input`'s edge cases

**Desktop verification:** ran `cargo run -p openlogi-desktop` against real hardware
(MX Keys S, Signature M650) — the app launched without a panic and the Devices grid
rendered correctly. I was not able to get a screenshot of the DPI panel itself: this
is a Wayland session with no safe way to synthesize a click into the running window
from here (no `ydotool`/`wtype`; an `xdotool` attempt sent a global XTEST event that
moved focus elsewhere on the desktop instead of reaching the app, so I did not
pursue it further). Screenshot follow-up welcome if useful for review.

Fixes #1008

### CODE REVIEW

**Comment by ZeysMockingjay** [comment]
Why isn't this merging again?

**Comment by 4ni1ak** [comment]
Nothing blocking on the technical side — CI is fully green and it's mergeable. `mergeStateStatus: BLOCKED` here just means it's waiting on a maintainer review/approval, which isn't something I can push forward myself. @AprilNEA whenever you get a chance to look at this one.

File: crates/openlogi-desktop/src/features/pointer/dpi.rs [hunk 1]
```diff
@@ -355,6 +468,41 @@ fn slider_element(
...
+/// only way to reach an out-of-range number here is typing more digits than
+/// any real DPI needs.
+fn parse_dpi_input(text: &str) -> Option<Dpi> {
+    let text = text.trim();
+    if text.is_empty() {
+        return None;
+    }
+    text.parse::<u32>()
+        .ok()
+        .map(|value| Dpi::new(u16::try_from(value).unwrap_or(u16::MAX)))
```
[the remark this answers is not in the document]
Comment (4ni1ak) [reply crates/openlogi-desktop/src/features/pointer/dpi.rs]: Confirmed, already fixed in b3a23b5 — `parse_dpi_input` now matches `IntErrorKind::PosOverflow` on the intermediate `u32` parse and saturates to `u16::MAX`, same as the existing `u16` conversion path. Added a regression test (`saturates_a_value_that_overflows_the_intermediate_parse`) covering a digit string long enough to overflow `u32` itself.
